### PR TITLE
MSC2240: Room version 6

### PR DESCRIPTION
**Pull request by turt2live**
[Rendered](https://github.com/matrix-org/matrix-doc/blob/travis/msc/rooms-v6/proposals/2240-rooms-v6.md)

For links, here's the MSCs included:
* [MSC2209](https://github.com/matrix-org/matrix-doc/pull/2209) - Including notifications in power level auth rules.
* [MSC2432](https://github.com/matrix-org/matrix-doc/pull/2432) - Alias event authorisation and redaction.
* [MSC2540](https://github.com/matrix-org/matrix-doc/pull/2540) - Integers in canonical JSON compliance.

### CODE REVIEW

**Comment by ara4n** [comment]
should we try to squeeze #2244 into here too?

**Comment by richvdh** [comment]
> should we try to squeeze #2244 into here too?

To be honest, I think it's unwise to set expectations that a given feature will or will not be in a given room version, particularly while that feature is still very much in development, since we end up in a situation where we never release the features we have because we are continually waiting for the next one that is almost-not-quite-ready-to-release.

(The same is true of a number of the other MSCs listed, fwiw: the rate of progress on their implementation is currently very low, and I think it's premature to propose them for room v6)

Let's wait until we think we have some things actually ready for release in a new room version, and decide then whether it's worth defining the room version at that point, or waiting for more features.

**Comment by ara4n** [comment]
i guess i was coming at this from the perspective of: "here is the backlog of MSCs which require a new room version; when we next cut a room version we should decide which are stable enough to merit inclusion so we don't forget any" rather than "let's block v6 on #2244".

**Comment by richvdh** [comment]
fair enough. I think we're using the [unassigned-room-version](https://github.com/matrix-org/matrix-doc/issues?q=label%3Aunassigned-room-version) label for that though.

**Comment by turt2live** [comment]
This is now ready for general review with the inclusion of the 3 MSCs we aim to target. There's a bunch of MSCs which are excluded, but if they happen to get enough of an implementation in time then they can certainly be added.

File: proposals/2240-rooms-v6.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# MSC2240: Room Version 6
+
+A new room version, `6`, is proposed using [room version 5](https://matrix.org/docs/spec/rooms/v5.html) as a base
+and incorporating the following MSCs:
+
+* [MSC2209](https://github.com/matrix-org/matrix-doc/pull/2209) - Including notifications in power level auth rules.
+* [MSC2212](https://github.com/matrix-org/matrix-doc/pull/2432) - Alias event authorisation and redaction.
```
Comment (clokep) [inline proposals/2240-rooms-v6.md:7]: Should this be MSC2432?

Comment (turt2live) [reply proposals/2240-rooms-v6.md:7]: Oops, yes.

**Comment by anoadragon453** [comment]
This lgtm matches my expectation as well. Most people seem in agreement so:

@mscbot fcp merge

**Comment by turt2live** [comment]
@mscbot concern The MSCs aren't in FCP

**Comment by anoadragon453** [comment]
All listed MSCs have now entered FCP.

@mscbot resolve The MSCs aren't in FCP

**Comment by turt2live** [comment]
Spec PR (for when this leaves FCP): https://github.com/matrix-org/matrix-doc/pull/2563

**Pull request merged**